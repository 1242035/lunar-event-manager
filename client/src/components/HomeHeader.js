import React from 'react';

/**
 * The banner of the homepage.
 * @param{undefined}
 * @return{React.Component}
 */
const HomeHeader = () => (
  <div className="io-parallax">
    <h1>Get 25% 0ff Now!</h1><p>A Special Place for Everyone</p>
  </div>
);

export default HomeHeader;
