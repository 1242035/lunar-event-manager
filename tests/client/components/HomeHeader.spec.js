import React from 'react';
import { shallow } from 'enzyme';
import HomeHeader from '../../../client/src/components/HomeHeader';

describe('HomeHeader', () => {
  let wrapper;

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<HomeHeader />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render component', () => {
    wrapper = shallow(<HomeHeader />);
    expect(wrapper.length).toEqual(1);
  });
});
