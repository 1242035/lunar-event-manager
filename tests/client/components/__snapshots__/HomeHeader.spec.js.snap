// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeHeader Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Get 25% 0ff Now!
        </h1>,
        <p>
          A Special Place for Everyone
        </p>,
      ],
      "className": "io-parallax",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get 25% 0ff Now!",
        },
        "ref": null,
        "rendered": "Get 25% 0ff Now!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A Special Place for Everyone",
        },
        "ref": null,
        "rendered": "A Special Place for Everyone",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Get 25% 0ff Now!
          </h1>,
          <p>
            A Special Place for Everyone
          </p>,
        ],
        "className": "io-parallax",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get 25% 0ff Now!",
          },
          "ref": null,
          "rendered": "Get 25% 0ff Now!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A Special Place for Everyone",
          },
          "ref": null,
          "rendered": "A Special Place for Everyone",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
